test/TestCriterion.cpp

```diff
@@ -9,88 +9,42 @@ TEST(TestCriterion, TestBaseConstructorWithoutSpec) {
   std::ostringstream os;
   os << crit;
   EXPECT_EQ(os.str(),
-            "Criterion(id : test_no_spec, name : , direction : +, weight : 0)");
+            "Criterion(id : test_no_spec, direction : +, weight : 0)");
 }
 
 TEST(TestCriterion, TestBaseConstructor) {
   std::string id = "test";
-  std::string name = "criterion_test";
   int direction = -1;
   float weight = 0.4;
-  Criterion crit = Criterion(id, name, direction, weight);
+  Criterion crit = Criterion(id, direction, weight);
   std::ostringstream os;
   os << crit;
-  EXPECT_EQ(os.str(), "Criterion(id : test, name : criterion_test, direction : "
-                      "-, weight : 0.4)");
+  EXPECT_EQ(os.str(), "Criterion(id : test, direction : -, weight : 0.4)");
 }
 
 TEST(TestCriterion, TestConstructorByCopy) {
   std::string id = "test_copy";
-  std::string name = "criterion_by_copy";
   int direction = 1;
   float weight = 0.3;
-  Criterion crit = Criterion(id, name, direction, weight);
+  Criterion crit = Criterion(id, direction, weight);
 
   Criterion critCopied = Criterion(crit);
 
   std::ostringstream os;
   os << critCopied;
-  EXPECT_EQ(os.str(), "Criterion(id : test_copy, name : criterion_by_copy, "
-                      "direction : +, weight : 0.3)");
+  EXPECT_EQ(os.str(), "Criterion(id : test_copy, direction : +, weight : 0.3)");
 }
 
-TEST(TestCriterion, TestGetterSetter) {
-  std::string id = "test";
-  std::string name = "criterion_test";
-  int direction = -1;
-  float weight = 0.4;
-  Criterion crit = Criterion(id, name, direction, weight);
-  EXPECT_EQ("test", crit.getId());
-  EXPECT_EQ("criterion_test", crit.getName());
-  EXPECT_EQ(-1, crit.getDirection());
-  EXPECT_FLOAT_EQ(0.4, crit.getWeight());
-
-  crit.setId("test_set");
-  crit.setName("criterion_test_set");
-  crit.setDirection(1);
-  crit.setWeight(0.3);
-
-  EXPECT_EQ("test_set", crit.getId());
-  EXPECT_EQ("criterion_test_set", crit.getName());
-  EXPECT_EQ(1, crit.getDirection());
-  EXPECT_FLOAT_EQ(0.3, crit.getWeight());
-}
-
-TEST(TestCriterion, TestDirectionGenerator) {
-  std::string id = "test_direction";
+/**
+TEST(TestCriterion, TestRandomWeightGeneration) {
```

Comment:
I thinkg we could remove this test :)

Reply:
Lets leave it commented for now until we know if we go forward with a in-house random number generator